tests/functional/hooks/fixtures.py

```diff
@@ -0,0 +1,333 @@
+macros__before_and_after = """
+{% macro custom_run_hook(state, target, run_started_at, invocation_id) %}
+
+   insert into {{ target.schema }}.on_run_hook (
+        "state",
+        "target.dbname",
+        "target.host",
+        "target.name",
+        "target.schema",
+        "target.type",
+        "target.user",
+        "target.pass",
+        "target.port",
+        "target.threads",
+        "run_started_at",
+        "invocation_id"
+   ) VALUES (
+    '{{ state }}',
+    '{{ target.dbname }}',
+    '{{ target.host }}',
+    '{{ target.name }}',
+    '{{ target.schema }}',
+    '{{ target.type }}',
+    '{{ target.user }}',
+    '{{ target.get("pass", "") }}',
+    {{ target.port }},
+    {{ target.threads }},
+    '{{ run_started_at }}',
+    '{{ invocation_id }}'
+   )
+
+{% endmacro %}
+"""
+
+macros__hook = """
+{% macro hook() %}
+  select 1
+{% endmacro %}
+"""
+
+models__hooks = """
+select 1 as id
+"""
+
+models__hooks_configured = """
+{{
+    config({
+        "pre_hook": "\
+            insert into {{this.schema}}.on_model_hook (\
+                \\"state\\",\
```

Comment:
yep, this much escaping is ~~the~~ an answer

Reply:
hurts my soul but it's jinja so whatever.